src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -425,7 +426,19 @@ def add(val):
     return result
 
 
-def process_docstring(app, what, name, obj, options, lines):  # noqa: U100
+def format_default(app: Sphinx, default):
+    if not app.config.typehints_defaults:
+        return None
```

Comment:
If this is the case the method will never be called so I don't think we need this